src/AzSK.ADO/Framework/Abstracts/MetaInfoProvider.ps1

```diff
@@ -0,0 +1,284 @@
+            $qs = "?svcId={0}" -f $svcId
+            $rsrcList = $this.CallADOInfoAPI($qs);
+        }
+        else 
+        {
+            $this.FetchMappingFiles($ResourceTypeName);
+
+            $buildList = @{};
+            $releaseList = @{};
+            $svcConnList = @{};
+            $varGroupList = @{};
+            $agentPoolList = @{};
+
+            if ($this.buildSTDetails) {
+                $buildList = $this.buildSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.releaseSTDetails) {
+                $releaseList = $this.releaseSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.svcConnSTDetails) {
+                $svcConnList = $this.svcConnSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.agtPoolSTDetails) {
+                $agentPoolList = $this.agtPoolSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            }
+            if ($this.varGroupSTDetails) {
+                $varGroupList = $this.varGroupSTDetails.Data | Where-Object { ($_.serviceId -eq $svcId) -and ($_.projectName -eq $projectName) }
+            } 
+            
+            $rsrcList = @{
+                Builds = $buildList
+                Releases = $releaseList
+                SvcConns = $svcConnList
+                VarGroups = $varGroupList
+                AgentPools = $agentPoolList
+            }
+
+        }
+        return $rsrcList; 
+    }
+    
+    [PSObject] FetchResourceMappingWithServiceData($rscId, $projectName, $resourceTypeName)
+    {
+        $serviceTree = $null;
+        try 
+        {
+            #check if adoinfoapi is enabled in org-policy file 
+            if ($this.bUseADOInfoAPI -eq $true -or $this.CheckIsADOInfoAPIEnabled())
+            {
+                $qs = "?ResourceType=$resourceTypeName";
+                #call adoinfoapi only if STDetails files is not already loaded.
+                if ( ($resourceTypeName -eq "Build" -and !$this.buildSTDetails) -or ($resourceTypeName -eq "Release" -and !$this.releaseSTDetails) -or ($resourceTypeName -eq "ServiceConnection" -and !$this.svcConnSTDetails) -or ($resourceTypeName -eq "AgentPool" -and !$this.agtPoolSTDetails)  -or ($resourceTypeName -eq "VariableGroupp" -and !$this.varGroupSTDetails) ) {
+                    $rsrcList = $this.CallADOInfoAPI($qs);
+                    $apiReturnedResourceTypeName = $resourceTypeName+"s";
+                    if ($rsrcList -and ( [Helpers]::CheckMember($rsrcList, "$apiReturnedResourceTypeName") -and $rsrcList."$apiReturnedResourceTypeName") ) {
+                        $this.BindADOInfoAPIResponseToSTMappingFiles($rsrcList, $resourceTypeName);
+                    }
+                    else {
```

Comment:
the else condition would be redundant if there is no mapping file for a particular rsrc type.